src/MultiFormats.Tests/MultiAddressTest.cs

```diff
@@ -0,0 +1,437 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Net;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+
+namespace MultiFormats.Tests
+{
+    [TestClass]
+    public class MultiAddressTest
+    {
+        private const string Somewhere =
+            "/ip4/10.1.10.10/tcp/29087/ipfs/QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC";
+
+        private const string Nowhere = "/ip4/10.1.10.11/tcp/29087/ipfs/QmVcSqVEsvm5RR9mBLjwpb2XjFVn5bPdPL69mL8PH45pPC";
+
+        [TestMethod]
+        public void Parsing()
+        {
+            var a = new MultiAddress(Somewhere);
+            Assert.AreEqual(3, a.Protocols.Count);
+            Assert.AreEqual("ip4", a.Protocols[0].Name);
+            Assert.AreEqual("10.1.10.10", a.Protocols[0].Value);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make sure using this hardcoded IP address '10.1.10.10' is safe here.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)